### ISSUE TITLE
[Localization 3.1] Failed to resolve string if RootNamespace is different from AssemblyName

### ISSUE THREAD

**Issue by jeremyVignelles** [dotnet/aspnetcore#21590]
On .NET core 3.1, if the RootNamespace is different from the AssemblyName, the resources fails to resolve.
Related to : https://github.com/aspnet/AspNetCore/issues/10639

Discussion continued from : dotnet/extensions#2081 

@hishamco asked me to file a new bug report, so here it is:

> Here is a repro project: https://github.com/jeremyVignelles/repro-localization
> The relevant line is here : https://github.com/jeremyVignelles/repro-localization/blob/5f0fc80ac58a6321c78f4ce0e743635809f20889/TestLocalization/Program.cs#L14 . If I use the ResourceManagerStringLocalizerFactory from 3.1, it doesn't work as expected.

**Comment by anurse** [comment]
Transferring to dotnet/aspnetcore, since Localization lives there now.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
@ryanbrandenburg can you please look into this? Thanks!

**Comment by pranavkm** [comment]
@jeremyVignelles does specifying the `RootNamespaceAttribute` work: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/localization?view=aspnetcore-3.1#rootnamespaceattribute? If not, do you have a minimal app that you can share that exhibits this problem?

**Comment by ryanbrandenburg** [comment]
@pranavkm [they provided a repro app](https://github.com/jeremyVignelles/repro-localization) already. That repro app bears out that they have used the RootNamespaceAttribute. I'm fairly confident that they are correct that we need to backport the fix they linked from 5.0 into 3.1, will be doing an investigation to confirm and backport today.

**Comment by jeremyVignelles** [comment]
@ryanbrandenburg was too fast for me :)

Yes, I did put the attribute, and it didn't fix anything.

My understanding is that yes, it was fixed in .NET 5, because the current "fix" I have is to copy the .NET 5 `ResourceManagerStringLocalizerFactory`. [See here](https://github.com/jeremyVignelles/repro-localization/blob/5f0fc80ac58a6321c78f4ce0e743635809f20889/TestLocalization/Net5ResourceManagerStringLocalizerFactory.cs)

It just wasn't backported as far as I'm aware of.
